src/BlockTemplatesController.php

```diff
@@ -52,6 +64,30 @@ protected function init() {
 		add_filter( 'pre_get_block_file_template', array( $this, 'get_block_file_template' ), 10, 3 );
 		add_filter( 'get_block_templates', array( $this, 'add_block_templates' ), 10, 3 );
 		add_filter( 'current_theme_supports-block-templates', array( $this, 'remove_block_template_support_for_shop_page' ) );
+
+		if ( $this->package->is_experimental_build() ) {
+			add_action( 'after_switch_theme', array( $this, 'check_should_use_blockified_product_grid_templates' ), 10, 2 );
+		}
+	}
+
+	/**
+	 * Checks the old and current themes and determines if the "wc_blocks_use_blockified_product_grid_block_as_template"
+	 * option need to be updated accordingly.
+	 *
+	 * @param string    $old_name Old theme name.
+	 * @param \WP_Theme $old_theme Instance of the old theme.
+	 * @return void
+	 */
+	public function check_should_use_blockified_product_grid_templates( $old_name, $old_theme ) {
```

Comment:
I wonder if we can optimize by creating a more abstract method which accepts the `option` key so we are able to use it for future templates we are going to convert to block templates to avoid creating separate methods for each scenario?